### PR TITLE
Remove dependency on Raize's Folder Selection Dialog

### PR DESCRIPTION
**Pull request by vor0nwe**
IMO, the 'native' one looks nicer (on Windows 10 at least). I've configured it so there is an edit field present, and a button, so users can create a new folder.
The Raize components are not free, AFAIK.

### CODE REVIEW

**Comment by rpottsoh** [comment]
@vor0nwe thanks for taking an interest in Exercism and the Windows Installer for the Windows version of the CLI.

This is just a quick note to let you know that I am evaluating your suggested changes to the code.  I think you may be correct regarding the Raize component I am using.  That particular component pack was freely available to me based on the version of the Delphi compiler I am using.  The code in this repo should be able to be compiled with only freely available components.

File: Delphi/Project/Source/uInstallLocationFrm.pas [hunk 1]
```diff
@@ -60,10 +61,13 @@ function ShowInstallLocationForm(var aInstallInfo: TInstallInfo): TResultStatus;
 end;
 
 procedure TfrmInstallLocation.btnBrowseClick(Sender: TObject);
+var
+  folder: string;
 begin
-  if RzSelectFolderDialog1.Execute then
+  folder := fldLocation.Text;
+  if SelectDirectory('Select Install Location', '', Folder, [sdNewUI, sdNewFolder, sdShowEdit], Self) then
```
Comment (rpottsoh) [inline Delphi/Project/Source/uInstallLocationFrm.pas]: Please remove `sdShowEdit`.  The edit field is confusing and I don't think necessary.

File: Delphi/Project/Source/uConfigApiFrm.pas [hunk 2]
```diff
@@ -67,10 +68,13 @@ procedure TfrmConfigAPI.btnFinishClick(Sender: TObject);
 end;
 
 procedure TfrmConfigAPI.btnBrowseClick(Sender: TObject);
+var
+  folder: string;
 begin
-  if RzSelectFolderDialog1.Execute then
+  folder := fldSolutionLocation.Text;
+  if SelectDirectory('Select Solution Location', '', folder, [sdNewUI, sdNewFolder, sdShowEdit], Self) then
```
Comment (rpottsoh) [inline Delphi/Project/Source/uConfigApiFrm.pas]: Please remove `sdShowEdit`.

**Comment by vor0nwe** [comment]
I've removed the edit boxes. I personally prefer them, because I often copy/paste paths, and having the edit box present allows me to paste a path, and thereby navigate much more quickly (especially in deep paths which require a lot of clicking to get to). But I know it's not a very popular MO, and I can imagine it can look confusing.
Thank you for writing the application! :-)

**Comment by rpottsoh** [comment]
@vor0nwe thanks for explaining to me how the edit fields are helpful.  I can't disagree with you.  The problem I run into (I try to take on the role of a noob) if I just type something in the field and then hit enter the dialog returns with a not quite desired result.  So to try and make it more intuitive for everyone I'd rather force the user to navigate to their desired folder.  If you already have the location in mind you could just paste it into the field and not use the Browse button.

Thanks again, I will approve your changes.

**Comment by rpottsoh** [comment]
@vor0nwe if you come across any other issues like the one you raised here (no pun intended, please don't hesitate to bring it to my attention.

**Pull request merged**